docs/user_guide/ttl.md

```diff
@@ -14,10 +14,32 @@ store-version, after the push finishes, but before swapping the read traffic ove
 replay phase where the last N seconds of buffered real-time data is replayed. The replayed data gets overlaid on top of
 the data from the Full Push.
 
-This can be leveraged in order to achieve a TTL behavior. By scheduling periodic empty pushes and configuring how far 
-back to replay the real-time data (N), one can control the TTL parameters. The time to live is defined by N, which acts 
-as a "minimum TTL", while the "maximum TTL" is N + delay between each empty push. For example, if you schedule a daily 
-empty push, and N = 6 days, then the oldest data in your store will be at least 6 days old, and at most 7 days old.
+This can be leveraged in order to achieve a TTL behavior. Currently, there are two options that leverages this mechanism. 
+
+### Empty push
+By scheduling periodic empty pushes and configuring how far back to replay the real-time data (N), one can control the 
+TTL parameters. The time to live is defined by N, which acts as a "minimum TTL", while the "maximum TTL" is N + delay 
+between each empty push. For example, if you schedule a daily empty push, and N = 6 days, then the oldest data in your 
+store will be at least 6 days old, and at most 7 days old.
+
+The N is as known as the store-level rewind time, that's a part of hybrid store configuration.
+
+### [Experimental] Repush with TTL
+Similarly to empty push, the repush with TTL can also be configured and scheduled periodically to achieve the TTL behavior.
+The major differences between this and empty push are original data source and how real-time buffer is replayed.
+
+The empty push literally has no original data, i.e. starts from scratch, and then replay real-time buffer from past N days,
+whereas the repush scans through entries in the current version, as its original data, and evict stale records based on 
+the write timestamp, and finally replays the real-time buffer from past M days. The M is known as version-level rewind time,
+that's a configurable value, 1 day by default, and normally smaller than the N(store-level rewind time).
+
+This brings two major benefits:
+1. The version topic, which is the data origin of repush, enables log compaction, plus the replayed records are less than 
+empty push, so the total amount of entries produced to the new store version will be reduced.
```

Comment:
I understand what you mean here but it seems a bit difficult to describe. I also think it's a bit of a minor point given that even if log compaction was disabled, point 2 would still de-dupe the data and the end result would be the same as far as the servers are concerned. Point 1 makes the VPJ part a bit more efficient, but in the grand scheme of things, it is a bit of a detail. Maybe removing point 1 entirely and focusing on just 2 may be easier to understand.

Or you could keep it as two points, but much more concise, e.g.:

```
This brings two major benefits:
1. The repush job de-dupes writes to the same key, so that the servers need to ingest fewer events.
2. The repush job sorts the data, thus allowing servers to ingest in a more optimized way.
```